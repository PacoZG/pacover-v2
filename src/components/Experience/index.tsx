'use client'
import React, { useEffect } from 'react'
import WorkingExperience from '@/components/Experience/WorkingExperience'
import { useTranslations } from 'next-intl'

const Experience: () => React.JSX.Element = () => {
  const t = useTranslations('stickers')
  useEffect(() => {
    window.scrollTo(0, 0)
  }, [])

  return (
    <div className="page-layout">
      <div className="flex justify-center w-full py-8">
        <div className="sticky-header rounded-t-xl">{t('experience')}</div>
      </div>

      <WorkingExperience />
    </div>
  )
}

export default Experience
