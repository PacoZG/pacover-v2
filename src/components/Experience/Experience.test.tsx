import React from 'react'
import { render } from '@testing-library/react'
import { describe, test, expect, vi, beforeEach } from 'vitest'
import Experience from '@/components/Experience/index'

describe('Experience', () => {
  describe('given configured with dependencies', () => {
    beforeEach(() => {
      window.scrollTo = vi.fn()

      vi.mock('next-intl', () => ({
        useTranslations: () => (key: string) => key,
      }))
    })

    describe('Experience component', () => {
      test('matches the snapshot', () => {
        const { container } = render(<Experience />)

        expect(container).toMatchSnapshot()
      })
    })
  })
})
