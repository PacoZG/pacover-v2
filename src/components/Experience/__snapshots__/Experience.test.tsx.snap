// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Experience > given configured with dependencies > Experience component > matches the snapshot 1`] = `
<div>
  <div
    class="page-layout"
  >
    <div
      class="flex justify-center w-full py-8"
    >
      <div
        class="sticky-header rounded-t-xl"
      >
        experience
      </div>
    </div>
    <div
      class=" flex flex-col items-center justify-center bg-gray-400 dark:bg-gray-900 rounded-b-xl p-4 py-10 md:py-14 lg:py-16 transition duration-500 bg-opacity-70 dark:bg-opacity-70 min-h-auto"
    >
      <div
        class="experience-card"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <h1
              class="experience-card-title"
            >
              <a
                class="experience-card-link"
                experience-tool-tip-msg="tool-tip-msg"
                href="https://houston-inc.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                company4
              </a>
            </h1>
            <div
              class="experience-card-period"
            >
              period4
            </div>
          </div>
          <div
            class="experience-card-position"
          >
            position4
          </div>
        </div>
        <div
          class="experience-card-responsibilities"
        >
          <ul
            class="list-disc list-inside space-y-1"
          >
            <li
              class="text-gray-700 dark:text-gray-400"
            >
              responsabilities4
            </li>
          </ul>
        </div>
      </div>
      <div
        class="experience-card"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <h1
              class="experience-card-title"
            >
              <a
                class="experience-card-link"
                experience-tool-tip-msg="tool-tip-msg"
                href="https://www.hotelkamp.com/en/"
                rel="noopener noreferrer"
                target="_blank"
              >
                company3
              </a>
            </h1>
            <div
              class="experience-card-period"
            >
              period3
            </div>
          </div>
          <div
            class="experience-card-position"
          >
            position3
          </div>
        </div>
        <div
          class="experience-card-responsibilities"
        >
          <ul
            class="list-disc list-inside space-y-1"
          >
            <li
              class="text-gray-700 dark:text-gray-400"
            >
              responsabilities3
            </li>
          </ul>
        </div>
      </div>
      <div
        class="experience-card"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <h1
              class="experience-card-title"
            >
              company2
            </h1>
            <div
              class="experience-card-period"
            >
              period2
            </div>
          </div>
          <div
            class="experience-card-position"
          >
            position2
          </div>
        </div>
        <div
          class="experience-card-responsibilities"
        >
          <ul
            class="list-disc list-inside space-y-1"
          >
            <li
              class="text-gray-700 dark:text-gray-400"
            >
              responsabilities2
            </li>
          </ul>
        </div>
      </div>
      <div
        class="experience-card"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="flex flex-col"
          >
            <h1
              class="experience-card-title"
            >
              company1
            </h1>
            <div
              class="experience-card-period"
            >
              period1
            </div>
          </div>
          <div
            class="experience-card-position"
          >
            position1
          </div>
        </div>
        <div
          class="experience-card-responsibilities"
        >
          <ul
            class="list-disc list-inside space-y-1"
          >
            <li
              class="text-gray-700 dark:text-gray-400"
            >
              responsabilities1
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
